### PR TITLE
Add vanillajs-datepicker typings

### PR DESCRIPTION
**Pull request by nihaopaul**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/vanillajs-datepicker/Datepicker.d.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    static get locales(): any;
+
+    constructor(element: any, options?: any, rangepicker?: DateRangePicker);
+    element: any;
+    config: any;
+    _options: any;
+    inline: boolean;
+    inputField: any;
+    dates: any;
+    picker: any;
```
Comment (martin-badin) [inline types/vanillajs-datepicker/Datepicker.d.ts:17]:
```suggestion
    config: any;
    dates: any;
    editMode: boolean;
    element: Element;
    inline: boolean;
    inputField: any;
    picker: any;
    
    _options: object;
    _showing: boolean;
```

File: types/vanillajs-datepicker/Datepicker.d.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+     */
+    setDate(...args: any[]): void;
+    /**
+     * Update the selected date(s) with input field's value
+     * Not available on inline picker
+     *
+     * The input field will be refreshed with properly formatted date string.
+     *
+     */
+    update(options?: any): void;
```
Comment (martin-badin) [inline types/vanillajs-datepicker/Datepicker.d.ts:83]:
```suggestion
    update(options?: {clear: boolean, render: boolean}): void;
```

File: types/vanillajs-datepicker/Datepicker.d.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    /**
+     * Refresh the picker element and the associated input field
+     */
+    refresh(target?: string, forceRender?: boolean): void;
```
Comment (martin-badin) [inline types/vanillajs-datepicker/Datepicker.d.ts:87]:
```suggestion
    refresh(forceRender?: boolean): void;
    refresh(target?: 'picker' | 'input', forceRender?: boolean): void;
```

File: types/vanillajs-datepicker/Datepicker.d.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    get active(): boolean;
+
+    get pickerElement(): any;
+
+    setOptions(options: any): void;
+    /**
+     * Show the picker element
+     */
+    show(): void;
+    _showing: boolean;
```
Comment (martin-badin) [inline types/vanillajs-datepicker/Datepicker.d.ts:28]:
```suggestion
```

File: types/vanillajs-datepicker/Datepicker.d.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
+import DateRangePicker from "./DateRangePicker";
+
+export default class Datepicker {
+    static formatDate(date: Date | number, format: string, lang?: string): string;
+
+    static parseDate(dateStr: string | Date | number, format: string, lang?: string): number;
+
+    static get locales(): any;
```
Comment (martin-badin) [inline types/vanillajs-datepicker/Datepicker.d.ts:8]:
```typescript
interface Local {
    clear?: string;
    days: string[];
    daysMin:string[];
    daysShort: string[];
    months: string[];
    monthsShort: string[];
    titleFormat?: string;
    today: string;
    rtl?: boolean;
}
```
```suggestion
    static get locales(): Record<string: Local>;
```

File: types/vanillajs-datepicker/Datepicker.d.ts [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    config: any;
+    _options: any;
+    inline: boolean;
+    inputField: any;
+    dates: any;
+    picker: any;
+
+    get active(): boolean;
+
+    get pickerElement(): any;
```
Comment (martin-badin) [inline types/vanillajs-datepicker/Datepicker.d.ts:21]:
```suggestion
    get active(): boolean;
    get pickerElement(): Element | undefined;
```

File: types/vanillajs-datepicker/Datepicker.d.ts [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    /**
+     * Refresh the picker element and the associated input field
+     */
+    refresh(target?: string, forceRender?: boolean): void;
+    /**
+     * Enter edit mode
+     * Not available on inline picker or when the picker element is hidden
+     */
+    enterEditMode(): void;
+    editMode: boolean;
```
Comment (martin-badin) [inline types/vanillajs-datepicker/Datepicker.d.ts:93]:
```suggestion
```

File: types/vanillajs-datepicker/tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,23 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6", "DOM"
```
Comment (martin-badin) [inline types/vanillajs-datepicker/tsconfig.json:5]:
```suggestion
            "es6", "dom"
```

File: types/vanillajs-datepicker/Datepicker.d.ts [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+    inline: boolean;
+    inputField: any;
+    dates: any;
+    picker: any;
+
+    get active(): boolean;
+
+    get pickerElement(): any;
+
+    setOptions(options: any): void;
```
Comment (martin-badin) [inline types/vanillajs-datepicker/Datepicker.d.ts:23]:
```suggestion
    setOptions(options: object): void;
```

File: types/vanillajs-datepicker/Datepicker.d.ts [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+     * Enter edit mode
+     * Not available on inline picker or when the picker element is hidden
+     */
+    enterEditMode(): void;
+    editMode: boolean;
+    /**
+     * Exit from edit mode
+     * Not available on inline picker
+     */
+    exitEditMode(options?: any): void;
```
Comment (martin-badin) [inline types/vanillajs-datepicker/Datepicker.d.ts:98]:
```suggestion
    exitEditMode(options?: object): void;
```

**Review by martin-badin — CHANGES_REQUESTED** [review]

File: types/vanillajs-datepicker/DateRangePicker.d.ts [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    allowOneSidedRange?: boolean;
+}
+export default class DateRangePicker {
+    constructor(element: HTMLElement, options?: DateRangePickerOptions);
+    allowOneSidedRange: boolean;
+    element: HTMLElement;
+    inputs: HTMLElement[];
+    datepickers: Datepicker[];
+
+get dates(): Date[] | undefined;
```
Comment (peterblazejewicz) [inline types/vanillajs-datepicker/DateRangePicker.d.ts:14]:
```suggestion
	get dates(): Date[] | undefined;
```

File: types/vanillajs-datepicker/index.d.ts [hunk 12]
```diff
@@ -0,0 +1,9 @@
+// Type definitions for vanillajs-datepicker 1.1
```
Comment (peterblazejewicz) [inline types/vanillajs-datepicker/index.d.ts:1]:
```suggestion
// Type definitions for vanillajs-datepicker 1.2
```
is this 1.1 specific or should be aligned with latest published (1.2)?

**Comment by peterblazejewicz** [comment]
Does it support global script (with iife block)?
https://unpkg.com/browse/vanillajs-datepicker@1.2.0/rollup.config.js
Should this DT support global script usage?

File: types/vanillajs-datepicker/DateRangePicker.d.ts [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+    allowOneSidedRange?: boolean;
+}
+export default class DateRangePicker {
+    constructor(element: HTMLElement, options?: DateRangePickerOptions);
+    allowOneSidedRange: boolean;
+    element: HTMLElement;
+    inputs: HTMLElement[];
+    datepickers: Datepicker[];
+
+	get dates(): Date[] | undefined;
```
Comment (DangerBotOSS) [inline types/vanillajs-datepicker/DateRangePicker.d.ts:14]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

Comment (gabritto) [inline types/vanillajs-datepicker/DateRangePicker.d.ts:14]:
```suggestion
    get dates(): Date[] | undefined;
```
Please use spaces instead of tabs here.

**Review by gabritto — APPROVED** [review]

**Pull request merged**